src/enchcracker/SimpleRandom.java

```diff
@@ -15,6 +15,16 @@ private int next() {
         return (int)(seed >>> 17);
     }
 
+    // always used
+    public int next8() {
+        return (int)((8 * (long)next()) >> 31);
+    }
+
+    // specifically for 15 shelves
+    public int next8and16() {
+        return (int)((8 * (long)next()) >> 31) + (int)((16 * (long)next()) >> 31);
```

Comment:
Same here. `(next() >>> 28) + (next() >>> 27)`. Also the method name could be better, maybe `plus` instead of `and`.